Volunteer Hours/VolunteerHoursScript.js

```diff
@@ -0,0 +1,126 @@
+
+const form = document.getElementById("volunteer-form");
+const charityInput = document.getElementById("charity-name");
+const hoursInput = document.getElementById("hours-volunteered");
+const dateInput = document.getElementById("date-volunteered");
+const ratingInput = document.getElementById("experience-rating");
+
+form.addEventListener("submit", async (event) => {
+  event.preventDefault();
+  await clearErrorMessages();
+  let isFormValid = await validateForm();
+  if (isFormValid) {
+      collectData(
+          charityNameInput.value,
+          hoursInput.value,
+          dateInput.value,
+          ratingInput.value
+      );
+  }
+});
+
+
+async function validateForm() {
+  let isFormValid = false;
+
+  const charityNameValid = await validateCharityNameInput();
+  const hoursValid = await validateHoursInput();
+  const dateValid = await validateDateInput();
+  const ratingValid = await validateRatingInput();
+
+  if (charityNameValid && hoursValid && dateValid && ratingValid) {
+      isFormValid = true;
+  }
+
+  return isFormValid;
+}
+
```

Comment:
I did the exact same thing as you but i figured out you can remove the validateForm() function by just calling all the other validate input functions in the eventListener